pywinauto/hooks.py

```diff
@@ -0,0 +1,231 @@
+              191: 'Oem_2',
+              192: 'Oem_3',
+              219: 'Oem_4',
+              220: 'Oem_5',
+              221: 'Oem_6',
+              222: 'Oem_7',
+              1001: 'mouse left',  # mouse hotkeys
+              1002: 'mouse right',
+              1003: 'mouse middle',
+              1000: 'mouse move',  # single event hotkeys
+              1004: 'mouse wheel up',
+              1005: 'mouse wheel down',
+              1010: 'Ctrl',  # merged hotkeys
+              1011: 'Alt',
+              1012: 'Shift',
+              1013: 'Win'}
+
+    event_types = {0x100: 'key down',  # WM_KeyDown for normal keys
+                   0x101: 'key up',  # WM_KeyUp for normal keys
+                   0x104: 'key down',  # WM_SYSKEYDOWN, used for Alt key.
+                   0x105: 'key up',  # WM_SYSKEYUP, used for Alt key.
+                   }
+
+    def __init__(self):
+        self.handler = 0
+        self.pressed_keys = []
+        self.id = None
+
+    def hook(self, keyboard=True, mouse=False):
+        if not mouse and not keyboard:
+            return;
+
+        if keyboard:
+            def low_level_handler(code, event_code, kb_data_ptr):
+                key_code = kb_data_ptr[0];
+                current_key = self.ID2Key[key_code];
+
+                event_code = self.event_types[event_code];
+
+                if event_code == 'key down':
+                    self.pressed_keys.append(current_key)
+
+                if event_code == 'key up':
+                    self.pressed_keys.remove(current_key)
+
+                event = KeyboardEvent(current_key, event_code, self.pressed_keys)
+
+                if self.handler != 0:
+                    self.handler(event);
+
+                return windll.user32.CallNextHookEx(self.id, code, event_code, kb_data_ptr)
+
+            keyboard_pointer = create_pointer(low_level_handler);
+            windll.kernel32.GetModuleHandleW.restype = wintypes.HMODULE
+            windll.kernel32.GetModuleHandleW.argtypes = [wintypes.LPCWSTR]
+            self.id = windll.user32.SetWindowsHookExA(0x00D, keyboard_pointer, windll.kernel32.GetModuleHandleW(None),
+                                                      0)
+
+        if mouse:
+            def mouse_low_level_handler(code, event_code, kb_data_ptr):
+                if event_code != 512:
+                    current_key = self.MouseId2Key[event_code]
+                    event_code = self.MouseId2EventType[event_code]
+
+                    event = MouseEvent(current_key, event_code)
+                    if self.handler != 0:
+                        self.handler(event);
+
+                return windll.user32.CallNextHookEx(self.id, code, event_code, kb_data_ptr)
+
+            mouse_pointer = create_pointer(mouse_low_level_handler);
+            self.id = windll.user32.SetWindowsHookExA(0x0E, mouse_pointer, windll.kernel32.GetModuleHandleW(None), 0)
+
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.id)
```

Comment:
This is great for a peace of mind. But a user should have an ability to unhook the handler at the moment he wants. So if we have `hook()` method, why not have `unhook()` too?